core/extension/gdextension_manager.cpp

```diff
@@ -93,6 +86,13 @@ GDExtensionManager::LoadStatus GDExtensionManager::_unload_extension_internal(co
 		}
 	}
 
+	if (level >= 0) { // Already initialized up to some level.
+		// Deinitialize down from current level.
+		for (int32_t i = level; i >= GDExtension::INITIALIZATION_LEVEL_CORE; i--) {
+			p_extension->deinitialize_library(GDExtension::InitializationLevel(i));
+		}
+	}
```

Comment:
I know this code just moved, but something I noticed is the use of `level >= 0` (ordinal number), but then
`i >= GDExtension::INITIALIZATION_LEVEL_CORE` (constant, but with implicit knowledge that `CORE == 0`).

The `if` is not really necessary at all, is it?

Values are defined as follows:

https://github.com/godotengine/godot/blob/46c495ca21f40f57a7fb9c7cde6143738f1652d4/core/extension/gdextension_interface.h#L729-L735

Reply:
`GDExtensionManager::level` defaults to `-1`:

https://github.com/godotengine/godot/blob/46c495ca21f40f57a7fb9c7cde6143738f1652d4/core/extension/gdextension_manager.h#L38

... which is used to indicate that we haven't been initialized up to any level yet. So, I think this `if` is still necessary.

I don't know that this is done in the clearest possible way, but it's not something to address in this PR :-)

Reply:
But in the case of `level == -1`, this `for` loop:
```rs
for (int32_t i = level; i >= GDExtension::INITIALIZATION_LEVEL_CORE; i--)
```
would run as:
```rs
for (int32_t i = -1; i >= 0; i--)
```
i.e. never -- so I don't think the `if` is needed.

I agree it doesn't have to be part of this PR, just noticed it during review.

Reply:
Oh, I see what you're saying now! Yeah, I think you're right :-)

Let's leave it be in this PR, though